src/main/java/org/apache/sysds/runtime/matrix/data/FrameBlock.java

```diff
@@ -2147,8 +2166,45 @@ public static FrameMapFunction getCompiledFunction(String lambdaExpr) {
 		}
 	}
 
+	public static FrameMapDistFunction getCompiledFunction2Args(String lambdaExpr) {
+		// split lambda expression
+		String[] parts = lambdaExpr.split("->");
+		parts[0] = parts[0].replace("(", "");
```

Comment:
these tow lines could be replaced with the following 
parts[0] = parts[0].replaceAll("[()]", "");